lib/handlers/check_channel_connection_handler.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:async';
+
+import 'package:breez/bloc/account/account_bloc.dart';
+import 'package:breez/logger.dart';
+import 'package:breez/services/device.dart';
+import 'package:breez/services/injector.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/flushbar.dart';
+import 'package:flushbar/flushbar.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+class CheckChannelConnection {
+  static final _instance = CheckChannelConnection._internal();
+
+  factory CheckChannelConnection() => _instance;
+
+  CheckChannelConnection._internal();
+
+  Timer notReadyTimer;
+  Flushbar _flushbar;
+
+  void startListen(
+    BuildContext context,
+    AccountBloc accountBloc,
+  ) {
+    ServiceInjector().device.eventStream.distinct().listen((event) {
+      switch (event) {
+        case NotificationType.RESUME:
+          accountBloc.accountStream
+              .map((acc) => acc.readyForPayments)
+              .distinct()
+              .listen((ready) {
+            if (ready) {
+              _readyForPayments();
+              notReadyTimer?.cancel();
+            } else {
+              notReadyTimer = Timer(Duration(seconds: 30), () {
+                _notReadyForPayments(context);
+              });
+            }
+          });
+          break;
+        case NotificationType.PAUSE:
+          notReadyTimer?.cancel();
```

Comment:
To me it looks almost complete. Only missing cancelling the subscription on pause

Reply:
✅  @roeierez